R/effectivemass.R

```diff
@@ -217,10 +217,7 @@ bootstrap.effectivemass <- function(cf, type="solve") {
   stopifnot(inherits(cf, 'cf_boot'))
 
   ## number of measurements
-  N <- length(cf$cf[,1])
-  if(is.null(cf$cf)) {
-    N <- cf$N
-  }
+  N <- length(cf$cf0)
```

Comment:
this cannot be correct, the number of measurements is not the length of `cf0`

Reply:
The order of operations on the LHS is simply wrong and probably the right thing to do would be to inherit `cf$N` to begin with.

Reply:
Oh, you are right. It is the number of measurements and not the number of time slices.

But why does the effective mass need to know the number of measurements anyway?

Reply:
Oh, it just preserves it. I think it just does not belong in there, just as the original correlator should not be in it. At least we have “effectivemass *has* cf” here, unlike “matrixfit *is* cf”.

Reply:
but now `N` is not defined anymore? `N` is still part of the returned `list`, isn't it?